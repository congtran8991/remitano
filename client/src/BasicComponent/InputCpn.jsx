import React from "react";

const InputCpn = (props) => {
  const {
    value = "",
    placeholder = "",
    onChange = () => {},
    name = "",
    type = "text",
    style = {},
    dataTestId = "defaultTestInputId",
  } = props;
  return (
    <input
      data-testid={dataTestId}
      type={type}
      name={name}
      className="input-cpn"
      value={value}
      placeholder={placeholder}
      onChange={onChange}
      style={style}
    />
  );
};

export default InputCpn;
