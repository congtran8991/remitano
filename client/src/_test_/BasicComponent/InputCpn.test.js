
import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import InputCpn from "../../BasicComponent/InputCpn";


describe("ButtonCpn", () => {
  test("calls onClick handler when clicked", () => {
    render(<InputCpn dataTestId="defaultTestInputId"/>)
    const inputCpn = screen.getByTestId("defaultTestInputId");
    fireEvent.change(inputCpn)
  });
});
