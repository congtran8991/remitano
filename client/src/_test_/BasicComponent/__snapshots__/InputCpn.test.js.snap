// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonCpn Render ButtonCpn 1`] = `undefined`;
